tracing-opentelemetry-http/src/main/java/io/micronaut/tracing/opentelemetry/instrument/http/client/OpenTelemetryClientFilter.java

```diff
@@ -0,0 +1,131 @@
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.micronaut.tracing.opentelemetry.instrument.http.client;
+
+import io.micronaut.context.annotation.Requires;
+import io.micronaut.core.annotation.Nullable;
+import io.micronaut.core.async.publisher.Publishers;
+import io.micronaut.http.HttpResponse;
+import io.micronaut.http.MutableHttpRequest;
+import io.micronaut.http.annotation.Filter;
+import io.micronaut.http.filter.ClientFilterChain;
+import io.micronaut.http.filter.HttpClientFilter;
+import io.micronaut.tracing.opentelemetry.instrument.http.AbstractOpenTelemetryFilter;
+import io.micronaut.tracing.opentelemetry.instrument.util.OpenTelemetryExclusionsConfiguration;
+import io.opentelemetry.api.OpenTelemetry;
+import io.opentelemetry.api.trace.Span;
+import io.opentelemetry.api.trace.Tracer;
+import io.opentelemetry.context.Context;
+import io.opentelemetry.context.Scope;
+import io.opentelemetry.instrumentation.api.instrumenter.Instrumenter;
+import org.reactivestreams.Publisher;
+import org.reactivestreams.Subscriber;
+import org.reactivestreams.Subscription;
+
+import static io.micronaut.tracing.opentelemetry.instrument.http.client.OpenTelemetryClientFilter.CLIENT_PATH;
+
+/**
+ * An HTTP client instrumentation filter that uses Open Telemetry.
+ *
+ * @author Nemanja Mikic
+ */
+@Filter(CLIENT_PATH)
+@Requires(beans = Tracer.class)
+public class OpenTelemetryClientFilter extends AbstractOpenTelemetryFilter implements HttpClientFilter {
+
+    @SuppressWarnings("rawtypes")
+    private final Instrumenter<MutableHttpRequest, HttpResponse> instrumenter;
+
+    /**
+     * Initialize the open tracing client filter with tracer and exclusion configuration.
+     *
+     * @param openTelemetry    the openTelemetry
+     * @param exclusionsConfig The {@link OpenTelemetryExclusionsConfiguration}
+     * @param openTelemetryHttpClientConfig The {@link OpenTelemetryHttpClientConfig}
+     */
+    public OpenTelemetryClientFilter(OpenTelemetry openTelemetry,
+                                     @Nullable OpenTelemetryExclusionsConfiguration exclusionsConfig,
+                                     @Nullable OpenTelemetryHttpClientConfig openTelemetryHttpClientConfig) {
+        super(exclusionsConfig == null ? null : exclusionsConfig.exclusionTest());
+        MicronautHttpClientTelemetryBuilder micronautHttpClientTelemetryBuilder = new MicronautHttpClientTelemetryBuilder(openTelemetry);
+
+        if (openTelemetryHttpClientConfig != null) {
```

Comment:
Suggestion: `Objects.nonNull()`

Reply:
that wouldn't be much of an improvement, and `nonNull` is intended to be used as a `Predicate`, not standalone